client_test.go

```diff
@@ -38,6 +38,53 @@ func TestGetMarathonURL(t *testing.T) {
 	assert.Equal(t, endpoint.Client.GetMarathonURL(), endpoint.URL)
 }
 
+func TestApiRequest(t *testing.T) {
+	cases := []struct {
+		Auth     bool
+		Username string
+		Password string
+		Ok       bool
+	}{
+		{
+			Username: "bad",
+			Ok:       true,
+		},
+		{
+			Auth:     true,
+			Username: "bad",
+		},
+		{
+			Auth:     true,
+			Username: "test",
+			Password: "test",
+			Ok:       true,
+		},
+	}
+	for i, x := range cases {
+		var endpoint *endpoint
+		switch x.Auth {
```

Comment:
We could get rid of `Auth` and the switch statement altogether with the following adjustments:

- Always pass username and password to the configuration and `newFakeAuthMarathonEndpoint`.
- Change `basicAuthMiddleware` to short-circuit / ignore authentication if either username and password are empty.

Note that the latter actually reflects [Marathon's real-world behavior](https://github.com/gambol99/go-marathon/pull/142#issuecomment-221151346): One must provide both a non-empty username and password in the server configuration, or otherwise end up with Marathon not doing any authentication at all.